pilot/pkg/model/authentication.go

```diff
@@ -125,7 +125,14 @@ func ConstructSdsSecretConfig(name, sdsUdsPath string, useK8sSATrustworthyJwt, u
 		gRPCConfig.CallCredentials = constructgRPCCallCredentials(K8sSATrustworthyJwtFileName, k8sSAJwtTokenHeaderKey)
 	} else if useK8sSANormalJwt {
```

Comment:
Much better - however... Will this custom 'sds token path' apply only if 'useK8SSANormalJwt' is used ? If I understand it correctly, the token may as well be a trustworthy JWT issued by Vault or someone else. 

Can we instead move this at the top - so if the metadata is set, it is used ? 

Most likely the correct behavior here is that if the App is passing a SdsTokenPath, we should generate SDS config overriding the global setting ( which will also help with incremental adoption )

Reply:
Done: moved SDS token path to the top so if the metadata is set, it is used.